### PR TITLE
Improve blob storage handling etcd backup

### PR DESCRIPTION
**Pull request by mohi-kalantari**
Don't start EtcdBackups until cluster is Ready

A new condition is added to wait for the ready cluster to start capturing the backups.

Wait for available blob storage in EtcdBackupNexus

Previously, e2e tests would fail because EtcdBackupNexus was created before MinioCluster. This caused the model to cache a nil blob storage value, leading to a terminal hang.

This commit introduces a wait loop to ensure blob storage is available before proceeding. Additionally, negative caching has been disabled to prevent disruptions to the e2e test flow.

Specs are also democked

### CODE REVIEW

File: model/kubernetes/kubernetes_etcd_backup.rb [hunk 1]
```diff
@@ -35,8 +36,7 @@ def next_backup_time
   end
 
   def blob_storage
-    return @blob_storage if defined?(@blob_storage)
-    @blob_storage = MinioCluster[project_id: Config.postgres_service_project_id, location_id: location.id]
+    @blob_storage ||= MinioCluster[project_id: Config.postgres_service_project_id, location_id: location.id]
```
Comment (jeremyevans) [inline model/kubernetes/kubernetes_etcd_backup.rb]: You are sure that you don't want to cache negative lookup? There are cases where you don't, but if so, there should be a commit explaining why negative lookup is not cached.

**Review by jeremyevans — COMMENTED** [review]
Looks mostly good.

File: model/kubernetes/kubernetes_etcd_backup.rb (hunk 1 above)
Comment (mohi-kalantari) [reply model/kubernetes/kubernetes_etcd_backup.rb]: Yep. Most of the times, the MinioCluster exists and there is no need for a negative cache but in the case of e2e tests, a MinioCluster is created at the same time the Backup is created and it would hang on a nil blob storage forever

Comment (mohi-kalantari) [reply model/kubernetes/kubernetes_etcd_backup.rb]: I updated the commit message

Comment (jeremyevans) [reply model/kubernetes/kubernetes_etcd_backup.rb]: I don't understand why this would hang on `nil` blob storage forever with negative lookup caching. The caching is per model instance, and each scheduled `respirate` strand run should use a new model instance. Did you test without this specific change (with the other changes in the PR) and confirm that it hangs forever? If so, can you point me to the failing E2E test run? If not, I recommend removing this particular change (and the related model spec change), and running E2E tests. I reviewed the E2E test result in this PR, which failed, and includes this change, so I'm not sure how the conclusion was reached that this change fixed an issue.

I guess we expect the cluster to always exist in production settings, so negative lookup caching is not that important. Removing it results in two separate queries for the `MinioCluster` in `wait` if the cluster doesn't exist, but maybe no effect other than that. Still, using an incorrect caching pattern should generally be avoided, as bad patterns tend to proliferate .

Comment (mohi-kalantari) [reply model/kubernetes/kubernetes_etcd_backup.rb]: got it thanks for the explanation. I reverted the negative cache.

**Comment by mohi-kalantari** [comment]
It failed because of this test
Test::GithubRunner.check_test_runs

re running the e2e tests

**Review by jeremyevans — COMMENTED** [review]
Looks good to me.

**Comment by aamederen** [comment]
LGTM as well, just waiting for a successful E2E run involving PG and K8s together

**Comment by mohi-kalantari** [comment]
The e2e pipeline only including k8s and postgres is passed

https://github.com/ubicloud/ubicloud/actions/runs/21221390844
